### PR TITLE
bulkio: Create new s3 client when accessing s3.

### PR DESCRIPTION
**Pull request by miretskiy**
Informs #56536

Create new s3 client whenever using external storage
to access s3 buckets.  This change reduces the scope
of lifetime of the s3 client so that implicit authentication
credentials with temporary session tokens get reloaded.

Release Notes: None

### CODE REVIEW

**Comment by miretskiy** [comment]
@dt Can you take a look?

File: pkg/storage/cloudimpl/s3_storage.go [hunk 1]
```diff
@@ -175,11 +162,29 @@ func MakeS3Storage(
...
 
+func (s *s3Storage) newS3Client(ctx context.Context) (*s3.S3, error) {
+	sess, err := session.NewSessionWithOptions(s.opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "new aws session")
+	}
+	if s.conf.Region == "" {
+		if err := delayedRetry(ctx, func() error {
+			var err error
+			s.conf.Region, err = s3manager.GetBucketRegion(ctx, sess, s.conf.Bucket, "us-east-1")
```
Comment (dt) [inline pkg/storage/cloudimpl/s3_storage.go:178]: this only runs the first time, since it mutates s.conf.Region so it's memoized for future calls?

**Review by dt — APPROVED** [review]

File: pkg/storage/cloudimpl/s3_storage.go (hunk 1 above)
Comment (miretskiy) [reply pkg/storage/cloudimpl/s3_storage.go:178]: Yup -- I really can't see region changing for a bucket.

**Pull request merged**